nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -208,7 +208,7 @@ protected void preProcessConfiguration(final Configuration config, final Process
         if (umaskProp.isSet()) {
             dfsUmask = Short.parseShort(umaskProp.getValue(), 8);
         } else {
-            dfsUmask = FsPermission.DEFAULT_UMASK;
+            dfsUmask = (short)config.getInt(FsPermission.UMASK_LABEL, FsPermission.DEFAULT_UMASK);
```

Comment:
This line should be simplified to:
```java
dfsUmask = FsPermission.getUMask(config).toShort();
```
This will handle retrieving the value of `fs.permissions.umask-mode` from the config or fall back to FsPermission.DEFAULT_UMASK if the property is missing from the config.  Please see: https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/permission/FsPermission.java#L305

Reply:
@jtstorck thank you for your review! I made those changes.